waku/v2/waku_archive/archive.nim

```diff
@@ -70,15 +74,57 @@ type
     validator: MessageValidator
     retentionPolicy: RetentionPolicy
 
+# these are defined few lines below. Just declaring to use them in 'new'.
+proc setupWakuArchiveDriver(url: string, vacuum: bool, migrate: bool):
+                            Result[ArchiveDriver, string]
+proc validateStoreMessageRetentionPolicy(val: string):
+                                         Result[string, string]
+proc setupWakuArchiveRetentionPolicy(retentionPolicy: string):
+                                     Result[RetentionPolicy, string]
+
 proc new*(T: type WakuArchive,
-          driver: ArchiveDriver,
-          validator = none(MessageValidator),
-          retentionPolicy = none(RetentionPolicy)): T =
-  WakuArchive(
-    driver: driver,
-    validator: validator.get(nil),
-    retentionPolicy: retentionPolicy.get(nil)
-  )
+          storeMessageDbUrl: string,
+          storeMessageDbVacuum: bool = false,
+          storeMessageDbMigration: bool = false,
+          storeMessageRetentionPolicy: string = "none"):
```

Comment:
I disagree that the Archive Driver should have any knowledge about underlying implementations. An Archive does not even have to be a DB at all - it could simply be an internal memory queue. The point of having a separate archive driver is exactly so that applications can set up and provide their own archive backend to the generic archive driver, with no information about application specific choices (do I have harddrives? do I have my own proprietary DB?) being passed down.